crates/core/src/db/datastore/locking_tx_datastore/datastore.rs

```diff
@@ -681,6 +681,10 @@ pub(super) fn record_metrics(
                 .inc_by(deletes.len() as u64);
         }
     }
+
+    if let Some(committed_state) = committed_state {
```

Comment:
In the code immediately above this, we are also updating a bunch of stats (counters and table size gauges). We should probably unify those at some point.

Reply:
I've added a TODO comment. For now I think it would be a hassle, since `report_data_size` is accessing private members of the committed state.